File: languages/php/prime.php
<?php
function is_prime($n) {
    if ($n <= 1) {
        return false;
    }
    if ($n == 2) {
        return true;
    }
    if ($n % 2 == 0) {
        return false;
    }

    for ($i = 3; $i <= sqrt($n); $i += 2) {
        if ($n % $i == 0) {
            return false;
        }
    }
    return true;
}

echo "Enter a number to check if it's prime: ";
$number = intval(trim(fgets(STDIN)));

if (is_prime($number)) {
    echo "$number is a prime number.\n";
} else {
    echo "$number is not a prime number.\n";
}
?>

File: languages/scala/Prime.scala
object Prime {
  def isPrime(n: Int): Boolean = {
    if (n <= 1) false
    else if (n == 2) true
    else !(2 to math.sqrt(n).toInt).exists(x => n % x == 0)
  }

  def main(args: Array[String]): Unit = {
    println("Enter a number to check if it's prime: ")
    val input = scala.io.StdIn.readInt()
    if (isPrime(input))
      println(s"$input is a prime number.")
    else
      println(s"$input is not a prime number.")
  }
}
